stdlib/ext/ext-test.ext-ocaml.mc

```diff
@@ -252,18 +252,18 @@ let extTestMap =
       impl
       {
         ident = "List.map",
-        ty = tyarrows_ [tyarrow_ (tyvar_ "a") (tyvar_ "b"),
-                        otylist_ (tyvar_ "a"),
-                        otylist_ (tyvar_ "b")]
+        ty = tyarrows_ [tyarrow_ (tycon_ "a") (tycon_ "b"),
```

Comment:
Yeah, it seems so. The previous `TyVar` was used both as type variables and type constructors,so just replacing them all with `TyCon` was a bit sloppy of me. There's a number other type signatures as well in e.g. `const-types.mc` which I haven't updated yet, which have `TyUnknown` instead of type variables. I'll try to fix the stuff directly touched by this PR, but leave the rest for a new one.